src/ui/tokens/update_token_config.rs

```diff
@@ -87,29 +86,204 @@ impl UpdateTokenConfigScreen {
 
         let mut error_message = None;
 
-        let group = match identity_token_info
-            .token_config
-            .manual_minting_rules()
-            .authorized_to_make_change_action_takers()
-        {
+        // Initialize with no group - will be set when user selects a change item
+        let group = None;
+
+        // Initialize as false - will be updated when group is determined
+        let is_unilateral_group_member = false;
+
+        // Attempt to get an unlocked wallet reference
+        let selected_wallet = get_selected_wallet(
+            &identity_token_info.identity,
+            None,
+            possible_key.as_ref(),
+            &mut error_message,
+        );
+
+        let data_contract_option = app_context
+            .get_contract_by_id(&identity_token_info.data_contract.contract.id())
+            .unwrap_or_default();
+
+        Self {
+            identity_token_info: identity_token_info.clone(),
+            data_contract_option,
+            backend_message: None,
+            update_status: UpdateTokenConfigStatus::NotUpdating,
+            app_context: app_context.clone(),
+            change_item: TokenConfigurationChangeItem::TokenConfigurationNoChange,
+            update_text: "".to_string(),
+            text_input_error: "".to_string(),
+            signing_key: possible_key,
+            public_note: None,
+
+            authorized_identity_input: None,
+            authorized_group_input: None,
+
+            selected_wallet,
+            wallet_password: String::new(),
+            show_password: false,
+            error_message,
+
+            identity: identity_token_info.identity,
+            group,
+            is_unilateral_group_member,
+            group_action_id: None,
+        }
+    }
+
+    fn determine_group_for_change_item(
+        change_item: &TokenConfigurationChangeItem,
+        identity_token_info: &IdentityTokenInfo,
+    ) -> (Option<(GroupContractPosition, Group)>, Option<String>) {
+        let mut error_message = None;
+
+        let authorized_action_takers = match change_item {
+            TokenConfigurationChangeItem::Conventions(_) => identity_token_info
+                .token_config
+                .conventions_change_rules()
+                .authorized_to_make_change_action_takers(),
+            TokenConfigurationChangeItem::ConventionsControlGroup(_)
+            | TokenConfigurationChangeItem::ConventionsAdminGroup(_) => identity_token_info
+                .token_config
+                .conventions_change_rules()
```

Comment:
it's the same as above (Conventions, ConventionsControlGroup,ConventionsAdminGroup all return the same result), please double-check and consolidate into one match case if this is correct.

Also applies to matches below.